roxie/ccd/ccdprotocol.cpp

```diff
@@ -1739,9 +1739,12 @@ class RoxieSocketWorker : public ProtocolQueryWorker
         unsigned remainingHttpConnectionRequests = global->maxHttpConnectionRequests ? global->maxHttpConnectionRequests : 1;
         unsigned readWait = WAIT_FOREVER;
 
+        SpanTimeStamp spanStartTimeStamp(true);
         Owned<IHpccProtocolMsgContext> msgctx = sink->createMsgContext(startTime);
 
 readAnother:
+        std::map<std::string, CCycleTimer> phaseTimers;
+        std::map<std::string, stat_type> phaseDurations;
```

Comment:
If you were to use a general purpose mechanism it should be a CRuntimeStatisticCollection